### PR TITLE
Add all variants of the Roboto font as assets to Material Design apps

### PR DESCRIPTION
**Pull request by jason-simmons**

### CODE REVIEW

**Comment by Hixie** [comment]
lgtm assuming that new fonts.zip has all the new fonts (i didn't check)

File: packages/flutter_tools/lib/src/flx.dart [hunk 1]
```diff
@@ -283,7 +288,8 @@ Future<int> assemble({
 
   zipBuilder.addEntry(_createAssetManifest(assets));
 
-  ZipEntry fontManifest = _createFontManifest(manifestDescriptor, usesMaterialDesign ? _getMaterialFonts() : null);
+  ZipEntry fontManifest = _createFontManifest(manifestDescriptor,
+      usesMaterialDesign ? _materialFontsManifest : null);
```
Comment (abarth) [inline packages/flutter_tools/lib/src/flx.dart]: We also want to make this conditional on whether we're building an FLX for iOS or for Android.  On Android, these fonts are already on the system so we don't need to pay the cost of including them in the FLX.

File: packages/flutter_tools/schema/material_fonts.json [hunk 2]
```diff
@@ -0,0 +1,97 @@
+[
```
Comment (abarth) [inline packages/flutter_tools/schema/material_fonts.json]: Just a nit pick, but I probably would have used yaml here for easy of copy/pasting between this file and flutter.yaml.

**Comment by jason-simmons** [comment]
Changed the manifest to YAML and gated this behind a flag

File: packages/flutter_tools/lib/src/commands/build_flx.dart [hunk 3]
```diff
@@ -23,6 +23,7 @@ class BuildFlxCommand extends FlutterCommand {
     argParser.addOption('snapshot', defaultsTo: defaultSnapshotPath);
     argParser.addOption('depfile', defaultsTo: defaultDepfilePath);
     argParser.addOption('working-dir', defaultsTo: defaultWorkingDirPath);
+    argParser.addFlag('include-roboto-fonts', defaultsTo: false);
```
Comment (abarth) [inline packages/flutter_tools/lib/src/commands/build_flx.dart]: I would have expected this to default to true and for the APK building to override this to false.

File: packages/flutter_tools/lib/src/flx.dart [hunk 4]
```diff
@@ -197,7 +214,8 @@ Future<int> build(
   String depfilePath: defaultDepfilePath,
   String privateKeyPath: defaultPrivateKeyPath,
   String workingDirPath: defaultWorkingDirPath,
-  bool precompiledSnapshot: false
+  bool precompiledSnapshot: false,
+  bool includeRobotoFonts: false
```
Comment (abarth) [inline packages/flutter_tools/lib/src/flx.dart]: ditto.

**Comment by abarth** [comment]
LGTM

File: packages/flutter_tools/lib/src/commands/build_flx.dart (hunk 3 above)
Comment (jason-simmons) [inline packages/flutter_tools/lib/src/commands/build_flx.dart]: done

File: packages/flutter_tools/lib/src/flx.dart (hunk 4 above)
Comment (jason-simmons) [inline packages/flutter_tools/lib/src/flx.dart]: done

**Comment by abarth** [comment]
LGTM

**Pull request merged**